pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSegmentUploadRestletResource.java

```diff
@@ -414,13 +415,23 @@ private String getDownloadUri(FileUploadType uploadType, Series headers, Represe
   })
   @Responses({
       @Response(statusCode = "200", description = "The segment was successfully uploaded"),
+      @Response(statusCode = "403", description = "Forbidden operation typically because it exceeds configured quota"),
       @Response(statusCode = "500", description = "There was an error when uploading the segment")
   })
   private Representation uploadSegment(File indexDir, File dataFile, String downloadUrl)
       throws ConfigurationException, IOException, JSONException {
     final SegmentMetadata metadata = new SegmentMetadataImpl(indexDir);
     final File tableDir = new File(baseDataDir, metadata.getTableName());
     File segmentFile = new File(tableDir, dataFile.getName());
+    StorageQuotaChecker.QuotaCheckerResponse quotaResponse = checkStorageQuota(indexDir, metadata);
+    if (!quotaResponse.isSegmentWithinQuota) {
+      // this is not an "error" hence we don't increment segment upload errors
+      setStatus(Status.CLIENT_ERROR_FORBIDDEN);
+      StringRepresentation repr = new StringRepresentation("{\"error\" : \"" + quotaResponse.reason + "\"}");
```

Comment:
Looks like you have this info in quota checker response, so maybe just log that as in INFO?

Basically, log all info that we need to explain why quota exceeded.thanks